src/System.Drawing.Common/src/System/Drawing/Font.Unix.cs

```diff
@@ -44,161 +44,21 @@ namespace System.Drawing
-                    break;
-                case GraphicsUnit.Pixel:
-                case GraphicsUnit.World:
-                    inchs = nSrc / Graphics.systemDpiX;
-                    break;
-                case GraphicsUnit.Point:
-                    inchs = nSrc / 72f;
-                    break;
-                default:
-                    throw new ArgumentException("Invalid GraphicsUnit");
-            }
-
-            switch (toUnit)
+            if (hfont == IntPtr.Zero)
             {
-                case GraphicsUnit.Display:
-                    nTrg = inchs * 75;
-                    break;
-                case GraphicsUnit.Document:
-                    nTrg = inchs * 300;
-                    break;
-                case GraphicsUnit.Inch:
-                    nTrg = inchs;
-                    break;
-                case GraphicsUnit.Millimeter:
-                    nTrg = inchs * 25.4f;
-                    break;
-                case GraphicsUnit.Pixel:
-                case GraphicsUnit.World:
-                    nTrg = inchs * Graphics.systemDpiX;
-                    break;
-                case GraphicsUnit.Point:
-                    nTrg = inchs * 72;
-                    break;
-                default:
-                    throw new ArgumentException("Invalid GraphicsUnit");
+                throw new ArgumentException(SR.GdiplusInvalidParameter);
             }
-        }
 
-        void setProperties(FontFamily family, float emSize, FontStyle style, GraphicsUnit unit, byte charSet, bool isVertical)
-        {
-            _name = family.Name;
-            _fontFamily = family;
-            _size = emSize;
-
-            // MS throws ArgumentException, if unit is set to GraphicsUnit.Display
-            _unit = unit;
-            _style = style;
-            _gdiCharSet = charSet;
-            _gdiVerticalFont = isVertical;
-
-            unitConversion(unit, GraphicsUnit.Point, emSize, out _sizeInPoints);
-
-            _bold = _italic = _strikeout = _underline = false;
-
-            if ((style & FontStyle.Bold) == FontStyle.Bold)
-                _bold = true;
-
-            if ((style & FontStyle.Italic) == FontStyle.Italic)
-                _italic = true;
-
-            if ((style & FontStyle.Strikeout) == FontStyle.Strikeout)
-                _strikeout = true;
-
-            if ((style & FontStyle.Underline) == FontStyle.Underline)
-                _underline = true;
-        }
-
-        public static Font FromHfont(IntPtr hfont)
-        {
             IntPtr newObject;
             FontStyle newStyle = FontStyle.Regular;
-            float newSize;
             SafeNativeMethods.LOGFONT lf = new SafeNativeMethods.LOGFONT();
 
-            // Sanity. Should we throw an exception?
-            if (hfont == IntPtr.Zero)
-            {
-                Font result = new Font("Arial", (float)10.0, FontStyle.Regular);
-                return (result);
-            }
```

Comment:
I assume this is covered by one of the tests you added / enabled?

Reply:
Yes. This now throws an ArgumentException - the diff is a bit confusing but if you look above you can see this